src/dev/flang/ast/Feature.java

```diff
@@ -2328,11 +2330,12 @@ AbstractType resultTypeIfPresent(Resolution res, List<AbstractType> generics)
           {
             res.resolveTypes(this);
           }
-        result = resultTypeRaw(res, generics);
+        result = resultTypeIfPresent(res);
         if (result instanceof UnresolvedType rt)
           {
             result = rt.visit(Feature.findGenerics,outer());
           }
+        result = result == null ? null : result.applyTypePars(this, generics);
```

Comment:
Can you add a NYI:-comment here to try to remove the call to `findGenerics` and uses `resultTypeIfPresent(res, generics)` instead of `resultTypeIfPresent(res)` followed by `applyTypePars`?

Reply:
My bad, I did not see that this was within `resultTypeIfPresent(res, generics)`. 
I am currently working on trying to clean up the `findGenerics`.